worlds/mlss/Rules.py

```diff
@@ -0,0 +1,571 @@
+from .Locations import all_locations, hidden
+from . import StateLogic
+
+if typing.TYPE_CHECKING:
+    from . import MLSSWorld
+
+
+def set_rules(world: "MLSSWorld", excluded):
+    for location in all_locations:
+        if "Digspot" in location.name:
+            if (world.options.skip_minecart and "Minecart" in location.name) or (
+                world.options.castle_skip and "Bowser" in location.name
+            ):
+                continue
+            if world.options.chuckle_beans == 0 or world.options.chuckle_beans == 1 and location.id in hidden:
+                continue
+            add_rule(
+                world.multiworld.get_location(location.name, world.player),
+                lambda state: StateLogic.canDig(state, world.player),
+            )
+        if "Beanstone" in location.name:
+            add_rule(
+                world.multiworld.get_location(location.name, world.player),
+                lambda state: StateLogic.canDig(state, world.player),
+            )
+        if "Shop" in location.name and "Coffee" not in location.name and location.name not in excluded:
+            forbid_item(world.multiworld.get_location(location.name, world.player), "Hammers", world.player)
+            if "Badge" in location.name or "Pants" in location.name:
+                add_rule(
+                    world.multiworld.get_location(location.name, world.player),
+                    lambda state: StateLogic.brooch(state, world.player) or StateLogic.rose(state, world.player),
+                )
+        if location.itemType != 0 and location.name not in excluded:
+            if "Bowser" in location.name and world.options.castle_skip:
+                continue
+            forbid_item(world.multiworld.get_location(location.name, world.player), "5 Coins", world.player)
+
+    if world.options.chuckle_beans == 2:
+        add_rule(
+            world.multiworld.get_location(LocationName.HoohooVillageSuperHammerCaveDigspot, world.player),
+            lambda state: StateLogic.canCrash(state, world.player) or StateLogic.super(state, world.player),
+        )
+        add_rule(
+            world.multiworld.get_location(LocationName.BeanbeanOutskirtsFarmRoomDigspot2, world.player),
+            lambda state: StateLogic.thunder(state, world.player),
+        )
+        add_rule(
+            world.multiworld.get_location(LocationName.BeanbeanOutskirtsFarmRoomDigspot3, world.player),
+            lambda state: StateLogic.thunder(state, world.player),
+        )
+        add_rule(
+            world.multiworld.get_location(LocationName.ChucklehuckWoodsWhiteFruitRoomDigspot3, world.player),
+            lambda state: StateLogic.canMini(state, world.player),
+        )
+        add_rule(
+            world.multiworld.get_location(LocationName.JokesEndJojoraRoomDigspot, world.player),
+            lambda state: StateLogic.canDash(state, world.player),
+        )
+
+    if world.options.chuckle_beans == 1 or world.options.chuckle_beans == 2:
```

Comment:
Two alternatives you could use. Feel free to just resolve this if you don't want to
```suggestion
    if world.options.chuckle_beans != 0:
```
```suggestion
    if world.options.chuckle_beans:
```